### PR TITLE
Implement device queue and device events endpoints for public api

### PR DESCRIPTION
**Pull request by simao**
Implement device queue and device events endpoints for public api
    
    `devices/:device-uuid/queue` returns the queue for a device
    
    `devices/:device-uuid/events` returns events received from a device

Fixed a bug when getting events by correlation id

Correct a few api definition typos

### CODE REVIEW

File: src/main/resources/api-definition.yml [hunk 1]
```diff
@@ -139,6 +194,82 @@ components:
...
+            type: object
+            required: [ecuId, name, timestamp]
+            properties:
+              ecuId:
+                type: string
+                format: string
+              name:
+                type: string
+                enum:
+                  - DownloadComplete
```
Comment (xaviergully) [inline src/main/resources/api-definition.yml]: Do all of these states exist on the portal ? 
I am just aware of downloadStart, downloadCompleted, installationStart, installationCompleted

**Comment by simao** [comment]
A previous version of this PR implemented a different API, the one discussed during planning:

/devices/:device-uuid/queue
/devices/:device-uuid/queue/:update-id/status

However, this is not possible to implement reliably using the data we have in our system. We don't have such a thing as "update status", so anything we would return to the client would be derived from the device events. But this would not be reliable, it's subject to change depending on the events the device pushes and the way device registry parses them, and it's not directly related to an update. Those events can, for example, not include a correlationId, which would mean we have no way to map that to an update.

For this reason, we implement a different API here:

```
/devices/:device-uuid/queue
/devices/:device-uuid/events?updateId=<updateId> # `updateId` is optional and is available in the `/queue` response
```

`/events` returns all events for a device. Optionally filtered for events that refer to an `updateId`. They are sorted by `deviceTime`, the timestamp the device attributed to the event. This is a similar API used by our UI, which then infers an update state, and this is what API clients will need to do. The last event in `/events` is the last known event for that update on that device and the api users will know what happened to the update last by checking the name of the event.

File: src/main/resources/api-definition.yml (hunk 1 above)
Comment (simao) [reply src/main/resources/api-definition.yml]: yes they exist.

**Comment by simao** [comment]
@Raigi new endpoints docs available here: http://api.porto.sit-ota.aws.in.here.com/docs You could start writing tests based on that schema if you are blocked

I will deploy to sit soon if @houcros and @jochenschneider don't raise any issues.

File: src/main/scala/com/advancedtelematic/ota/api_provider/http/ApiProvider.scala [hunk 2]
```diff
@@ -46,4 +46,47 @@ class ApiProvider(directorClient: DirectorClient)(implicit ec: ExecutionContext,
...
+        }
+
+      }.toVector.flatten
+    }
+  }
+
+  def findUpdateEvents(namespace: Namespace, deviceId: DeviceId, correlationId: Option[CorrelationId]): Future[ApiDeviceEvents] = async {
+    val indexedEvents = await(eventJournal.getIndexedEvents(deviceId, correlationId))
+
+    println(indexedEvents.map(_._2.eventType))
```
Comment (jochenschneider) [inline src/main/scala/com/advancedtelematic/ota/api_provider/http/ApiProvider.scala]: Leftover?

File: src/main/scala/com/advancedtelematic/ota/deviceregistry/DevicesResource.scala [hunk 3]
```diff
@@ -254,6 +255,8 @@ class DevicesResource(
         path("events") {
           import DevicesResource.EventPayloadDecoder
           (get & parameter('correlationId.as[CorrelationId].?)) { correlationId =>
+            // TODO: This should not return raw Events
+            // API should not return arbitrary json (`payload`) to the clients. This si why we index interesting events, so we can give this info to clients
```
Comment (jochenschneider) [inline src/main/scala/com/advancedtelematic/ota/deviceregistry/DevicesResource.scala]:
```suggestion
            // API should not return arbitrary json (`payload`) to the clients. This is why we index interesting events, so we can give this info to clients
```

Comment (jochenschneider) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/DevicesResource.scala]: Do you want to create a ticket for this?

File: src/test/scala/com/advancedtelematic/ota/api_provider/http/DeviceInfoResourceSpec.scala [hunk 4]
```diff
@@ -116,4 +125,136 @@ class DeviceInfoResourceSpec extends FunSuite with ResourceSpec with Eventually
...
+      "correlationId" -> campaignId01.toString().asJson
+    ).asJson
+
+    val payload02 = Map(
+      "ecu" -> ecuId.asJson,
+      "correlationId" -> campaignId02.toString().asJson
+    ).asJson
+
+    val event01 = Event(deviceId, UUID.randomUUID().toString, EventType("EcuInstallationStarted", 0), now, now, payload01)
+    new DeviceEventListener().apply(DeviceEventMessage(defaultNs, event01)).futureValue
```
Comment (jochenschneider) [inline src/test/scala/com/advancedtelematic/ota/api_provider/http/DeviceInfoResourceSpec.scala]: Put the DeviceEventListener into a variable and call it twice?

**Pull request merged**